internal/audit/hub/ingest.go

```diff
@@ -50,13 +52,36 @@ func NewIngestSyncer(logger *zap.Logger) (*Impl, error) {
 	}, nil
 }
 
-func (i *Impl) Sync(ctx context.Context, batch *logsv1.IngestBatch) error {
+func (i *Impl) Sync(ctx context.Context, batch *logsv1.IngestBatch) (err error) {
 	if len(batch.GetEntries()) > 0 {
-		if err := i.client.Ingest(ctx, batch); err != nil {
+		var backoff time.Duration
+		defer func() {
+			// if the server responds with a backoff, return it regardless of whether an error exists
+			if backoff != 0 {
+				err = ErrIngestBackoff{
+					underlying: err,
+					Backoff:    backoff,
+				}
+			} else if err != nil {
+				// Apply fixed default backoff for severe failures (typically network issues)
+				// where the server couldn't communicate a specific backoff duration.
+				// Using a conservative fixed backoff rather than exponential backoff as
+				// these errors are expected to be transient network issues and retrying
+				// too aggressively could exacerbate the problem (and because an exponential
+				// backoff here feels entirely overkill).
```

Comment:
I'm not totally convinced by this line of reasoning; the point of exponential backoff strategies is to mitigate exactly the issue you're describing, isn't it?

We can tune the parameters of the exponential backoff so that there is a sensible minimum interval, but in the face of an outage we don't want every PDP connected to Hub to be constantly retrying at the same rate, so an exponential backoff with jitter is definitely appropriate here.

Reply:
You're right. I'd estimated the maintenance complexity probably wasn't worth the effort given our current usage and mechanisms, but it's actually not that complex and is sensible. Our couple of cows today could become a thundering herd tomorrow 🙂